src/Domain/FunctionsOfTime/PiecewisePolynomial.hpp

```diff
@@ -108,9 +108,10 @@ class PiecewisePolynomial : public FunctionOfTime {
   // the values of that deriv order for all components.
   using value_type = std::array<DataVector, MaxDeriv + 1>;
 
-  std::vector<FunctionOfTimeHelpers::StoredInfo<MaxDeriv + 1>>
+  std::deque<FunctionOfTimeHelpers::StoredInfo<MaxDeriv + 1>>
       deriv_info_at_update_times_;
   double expiration_time_{std::numeric_limits<double>::lowest()};
+  size_t deriv_info_size_{};
```

Comment:
Does this have to be atomic?

Reply:
Hmmm yes probably. If I make it atomic, do I need to add in any locks or is that taken care of? Also, in the `stored_info_from_upper_bound` function, do I need to take a reference to the `size_t` then? Or is it alright to have it copy?

Reply:
the int is 64-bytes aligned, but if it doesn't end on a 64-byte boundary then you can get false sharing on the next cache line. That is, if the `atomic<int>` is 80 bytes then you want to align it at 64 bytes add pad it out to 128 bytes so the int is all in its own cache lines